robotidy/files.py

```diff
@@ -84,12 +84,16 @@ def get_gitignore(root: Path) -> PathSpec:
     return PathSpec.from_lines("gitwildmatch", lines)
 
 
-def path_is_excluded(
-    normalized_path: str,
-    pattern: Optional[Pattern[str]],
-) -> bool:
-    match = pattern.search(normalized_path) if pattern else None
-    return bool(match and match.group(0))
+def path_is_excluded(path: Path, exclude: Optional[Pattern[str]], extend_exclude: Optional[Pattern[str]]) -> bool:
+    def _path_is_excluded(
+        normalized_path: str,
+        pattern: Optional[Pattern[str]],
+    ) -> bool:
+        match = pattern.search(normalized_path) if pattern else None
+        return bool(match)
+
+    normalized_path = str(path)
+    return _path_is_excluded(normalized_path, exclude) or _path_is_excluded(normalized_path, extend_exclude)
```

Comment:
Instead of defining method in method we could define another method (ie "should_parse_path") and then check all conditions here:
```
def should_parse_path(...args...):
    if path.suffix not in INCLUDE_EXT:
        return False
    if gitignore is not None and gitignore.match_file(path):
        return False
    path_str = str(path)
    return not(path_is_excluded(path_str, exclude) or path_is_excluded(path_str, extend_exclude)
```
then replace lines 121 - 124 (in the original source) by calling this method and add in `get_paths` like you did here with `path_is_excluded`

Reply:
The con of this solution is that there will be no way to run file if it matches any excluder (be it directories like `.git` or file specified with `.gitignore` or `extend_exclude`). But I assume that if someone specified file in gitignore he/she rather don't want to transform this file anyway.

Reply:
perhaps a flag to turn off the exclude checking, such as `--no-exclude`?

Reply:
Good idea - I will not implement it now but I will keep it in mind (especially in case the need will arise for it)